src/filters/KalmanFilter.h

```diff
@@ -2,9 +2,12 @@
 
 #include <Eigen/Core>
 #include <Eigen/LU>
-#include <unsupported/Eigen/MatrixFunctions>
 
-template <int numStates, int numInputs> class KalmanFilter
+#include "KalmanFilterBase.h"
```

Comment:
I'm having some trouble understanding what KalmanFilter.h is for. It looks like it's used by PoseEstimatorLinear, the deprecated old version of the pose estimator. Is it used by anything else?

Also, I would think that a Kalman Filter is just a special case of an extended Kalman Filter, where the derivatives are all constant. Is the idea here just to keep the old code in case there's a bug in the new, more general implementation?

Reply:
Yeah, the KalmanFilter class is only used by the PoseEstimatorLinear.

During development/testing I was using the KF pose estimator as a baseline to compare the EKF against, so I needed the PoseEstimatorLinear to still exist. My reasoning for keeping it going forward is that it's working estimation code that's simpler than the EKF, so if the EKF turns out to be terrible on real robots, we can test it against the KF estimator.

Now that you mention it though, since the KF is just a special case of the EKF there might not be a point in keeping it, assuming the EKF implementation is lacking significant bugs. The KF is probably computationally cheaper, but I'm not sure if that's reason enough to keep it.